clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -120,7 +120,7 @@ class FunctionVisitor : public RecursiveASTVisitor<FunctionVisitor> {
     // Is cast compatible with LHS type?
     QualType SrcT = C->getSubExpr()->getType();
     QualType DstT = C->getType();
-    if (!isCastSafe(DstT, SrcT)) {
+    if (!isCastSafe(DstT, SrcT) && !Info.hasPersistentConstraints(C, Context)) {
```

Comment:
Can you remind me what the "persistent" aspects of a constraint set is? I thought that persistence was something that wasn't actually in play (there was a flag in `ProgramInfo` that wasn't used for anything)?

Reply:
Persistent means that the constraint variables for the expression have been saved in a set so that we can use them during rewriting. Since we save the constraints for expressions in this set when they're computed, it also acts a cache to avoid recomputing the constraints or an expression multiple times. In this case, I want to avoid adding an extra edge to WILD if we've generated constraints for the cast because extra edges from WILD dilute the root analysis weights.